python/lsst/pipe/base/graphBuilder.py

```diff
@@ -579,6 +579,9 @@ def connectDataIds(self, registry, collections, userQuery, externalDataId):
                     for datasetType in task.outputs:
                         ref = refsForRow[datasetType.name]
                         quantum.outputs[datasetType.name][ref.dataId] = ref
+            if n == 0:
+                for message in commonDataIds.explain_no_results():
+                    _LOG.warn(message)
```

Comment:
`warn()` is deprecated, better to use `warning()`

Reply:
Ack, missed this due to some kind of browser-refreshing misfire, and I just merged.  Will fix in my next pipe_base ticket.